src/Builder/OutputReader.php

```diff
@@ -0,0 +1,55 @@
+<?php
+/**
+ * @copyright 2018 Hostnet B.V.
+ */
+declare(strict_types=1);
+
+namespace Hostnet\Component\Resolver\Builder;
+
+use Hostnet\Component\Resolver\File;
+use Hostnet\Component\Resolver\Report\ReporterInterface;
+
+/**
+ * Helper class for buffering the build.js output and reporting it.
+ */
+class OutputReader
+{
+    private $reporter;
+    private $buffer = '';
+
+    public function __construct(ReporterInterface $reporter)
+    {
+        $this->reporter = $reporter;
+    }
+
+    public function append(string $content): void
+    {
+        $this->buffer .= $content;
+
+        // Now parse it line-by-line
+        while (false !== ($i = strpos($this->buffer, "\n"))) {
```

Comment:
This is needed in the case where the content didn't contain a new line. For instance:
```php
$reader->append("foo");
$reader->append("bar\nbaz");
```
After the first call the buffer will still contain `'foo'` as a string so it will be prepended to the next call.  In the next call the string `'foobar'` will be processed and the buffer still contains `'bar'`.

I cannot use explode, since I cannot detect correctly if the string had a remainder or just ended with a new line.